src/Visualization/Python/PlotEllipticConvergence.py

```diff
@@ -73,39 +75,71 @@ def plot_elliptic_convergence(
         else linear_residuals
     )[0]["Residual"].iloc[0]
     # Plot nonlinear solver residuals
-    if ax is not None:
-        plt.sca(ax)
+    if axes is None:
+        fig, (ax_residual, ax_time) = plt.subplots(
+            nrows=2, ncols=1, sharex=True, gridspec_kw={"height_ratios": [3, 1]}
+        )
+    else:
+        ax_residual, ax_time = axes
     if nonlinear_residuals is not None:
         m = 0
         for i, residuals in enumerate(nonlinear_residuals):
-            plt.plot(
+            ax_residual.plot(
                 cumulative_linsolv_iterations[m : m + len(residuals)],
                 residuals["Residual"] / norm,
                 color="black",
                 ls="dotted",
                 marker=".",
-                label="Nonlinear residual" if i == 0 else None,
+                label="Nonlinear solver" if i == 0 else None,
+            )
+            ax_time.plot(
+                cumulative_linsolv_iterations[m : m + len(residuals) - 1],
+                np.diff(residuals["Walltime"]),
```

Comment:
Check if the "Walltime" was dumped first. Otherwise this won't work on any existing ID runs. We can remove the check later once enough time has passed and all runs we are interested in output the walltime.

Reply:
Ok pushed a fixup